### PR TITLE
[libmt32emu] new port

### LINKED ISSUE #16813: [New Port Request]  libmt32emu
**Name**:  libmt32emu

**Description**: libmt32emu is a stand-alone C and C++ library within the MT-32 emulator Munt software package. It's used in several game packages including ScummVM and DOSBox Staging. 

**Source URL**: https://github.com/munt/munt/tree/master/mt32emu (the library sub-project)

**Project homepage**: https://github.com/munt/munt

**Helpful information:**

Munt is a Cmake project.

libmt32emu is a subdirectory (`src/mt32emu`) within the `Munt` repository.  We are only interested in this stand-alone library for vcpkg.

To build it, we suggest using options that produce a stand-alone library without dependencies, such as:

``` shell
cd src/mt32emu
cmake \
  -DCMAKE_INSTALL_PREFIX:PATH=C:/vcpkg/installed/<your target>
  -DCMAKE_BUILD_TYPE:STRING=Release \
  -Dlibmt32emu_WITH_INTERNAL_RESAMPLER=true \
  -Dlibmt32emu_SHARED=false \
  .
make
ls libmt32emu.a
make install
```

### PR DESCRIPTION
**Pull request by autoantwort**
**Describe the pull request**

- What does your PR fix? Fixes #16813

- Which triplets are supported/not supported? hopefully all

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)? yes

### CODE REVIEW

**Comment by electron271** [comment]
Looks good!

**Comment by electron271** [comment]
Not much of a coder, but this looks pretty good!

**Comment by kcgen** [comment]
Thanks for the fast and elegant support for this library @autoantwort!

The plan is to have MT-32 support included in our upcoming Dosbox Staging release, so can't wait to put this into our VisualStudio solution and start testing with it!

**Comment by dreamer** [comment]
Oh, nice - that was fast - I bet @sergm will want to know that the library is available in vcpkg repo :)

File: ports/libmt32emu/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}/mt32emu
+    PREFER_NINJA
+    OPTIONS
+        -Dlibmt32emu_SHARED:BOOL=${BUILD_SHARED}
+)
+
+vcpkg_install_cmake()
+
```
Comment (JackBoosY) [inline ports/libmt32emu/portfile.cmake:19]:
```suggestion
vcpkg_copy_pdbs()

```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: versions/l-/libmt32emu.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+  "versions": [
+    {
+      "git-tree": "dc09e3df6c7048a7256000ab16f79a2a7b4a2820",
```
Comment (JackBoosY) [inline versions/l-/libmt32emu.json:4]:
```suggestion
      "git-tree": "b60fd8501e86b53f2d3125d0f0a1a793ff5f9a60",
```

**Review by JackBoosY — APPROVED** [review]

File: versions/l-/libmt32emu.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  "versions": [
+    {
+      "git-tree": "b60fd8501e86b53f2d3125d0f0a1a793ff5f9a60",
+      "version": "2.4.2",
+      "port-version": 0
+    },
+    {
+      "git-tree": "09fab58bf66b5ea7d702dad8e96b3b57b852f4cc",
+      "version-date": "2021-03-21",
+      "port-version": 0
```
Comment (ras0219-msft) [inline versions/l-/libmt32emu.json:11]:
```suggestion
```

Each PR must only add one version of a library

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
LGTM once the 2021-03-21 is removed, thanks for the PR!

File: versions/l-/libmt32emu.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    }
+  ]
+}
```
Comment (JackBoosY) [inline versions/l-/libmt32emu.json]: Can you please remove this file and rerun the command `./vcpkg x-add-version libmt32emu` then commit the changes?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: versions/l-/libmt32emu.json (hunk 4 above)
Comment (autoantwort) [reply versions/l-/libmt32emu.json]: That is already fixed ??

**Review by JackBoosY — APPROVED** [review]

**Comment by kcgen** [comment]
@autoantwort , @JackBoosY , @ras0219-msft - the current PR was tested in DOSBox Staging.

The resulting `libmt32emu` package was installed via `vcpkg`, `mt32emu.lib` and `mt32emu-2.dll` werw linked into the binary, and is running as expected!  Thank you 👍

**Comment by kcgen** [comment]
Thanks for the review @ras0219-msft! 

> LGTM once the 2021-03-21 is removed, thanks for the PR!

Looks like @autoantwort removed this.

If there's additional testing or checks needed on the user-side, just shout.

**Comment by ras0219-msft** [comment]
Thanks everyone!

**Comment by kcgen** [comment]
@autoantwort , we caught the tail end of the 2.4-series of `libmt32emu` in vcpkg. 
v2.5.0 has just been tagged and confirmed good on our side, as a user of a library.
I'll open ticket to bump the vcpkg repo to use 2.5.0.

**Pull request merged**